src/js/game/buildings/analyzer.js

```diff
@@ -3,51 +3,120 @@ import { enumDirection, Vector } from "../../core/vector";
 import { enumLogicGateType, LogicGateComponent } from "../components/logic_gate";
 import { enumPinSlotType, WiredPinsComponent } from "../components/wired_pins";
 import { Entity } from "../entity";
-import { MetaBuilding } from "../meta_building";
+import { defaultBuildingVariant, MetaBuilding } from "../meta_building";
 import { GameRoot } from "../root";
 import { enumHubGoalRewards } from "../tutorial_goals";
 
-const overlayMatrix = generateMatrixRotations([1, 1, 0, 1, 1, 1, 0, 1, 0]);
-
 export class MetaAnalyzerBuilding extends MetaBuilding {
     constructor() {
         super("analyzer");
     }
 
-    getSilhouetteColor() {
-        return "#3a52bc";
+    /**
+     * @param {string} variant
+     */
+    getSilhouetteColor(variant) {
+        return MetaAnalyzerBuilding.silhouetteColors[variant]();
     }
 
     /**
      * @param {GameRoot} root
      */
     getIsUnlocked(root) {
-        return root.hubGoals.isRewardUnlocked(enumHubGoalRewards.reward_virtual_processing);
+        return this.getAvailableVariants(root).length > 0;

```

Comment:
ah yeah, I should have done this much ealier. Maybe we can do it for all buildings like this ?

Reply:
Sure

Reply:
I already is for all buildings, maybe make it default in metabuilding?

Reply:
I made it default in the metabuilding.